src/features/Chat/components/ActiveWindow/index.tsx

```diff
@@ -20,30 +24,58 @@ import {
   MOBILE_AND_TABLET_CONTENT_HEIGHT,
   palette,
 } from '@/components/constants';
-
 import Header from './Header';
 import MessageField from './MessageField';
 import MessageList from './MessageList';
 
 const ActiveWindow = () => {
   const { isTablet } = useGetLayoutMode();
   const dispatch = useAppDispatch();
+  const [sendMessage, { isLoading: isLoadingSendMessage }] =
+    useSendMessageMutation();
+  const userId = useAppSelector(selectUserId);
 
   const activeChat = useAppSelector(selectActiveChat);
   const defaultChat = useAppSelector(selectDefaultChat);
   const chat = activeChat ?? defaultChat;
 
-  useEffect(() => {
-    dispatch(setActiveChat(chat.buddyId));
-  }, [chat.buddyId]);
-
-  const [, { isLoading: isLoadingSendMessage }] = useSendMessageMutation();
+  const [message, setMessage] = useState('');
 
   const isLoadingMessages = useAppSelector(
     selectIsLoadingBuddyMessages(chat?.buddyId),
   );
+  const isLoading = isLoadingMessages || isLoadingSendMessage;
+
+  const [isLoadingNewMessage, setIsLoadingNewMessage] = useState(false);
+  const isSendingDisabled = isLoadingNewMessage || message.trim().length < 1;
+
+  const handleMessageSend = async () => {
+    if (!userId || isLoadingNewMessage) return;
 
-  const isLoading = isLoadingSendMessage || isLoadingMessages;
+    setIsLoadingNewMessage(true);
+    const trimmedMessage = message.trim();
+
+    try {
+      await sendMessage({
+        userId,
+        message: toSendMessage(chat.buddyId, userId, trimmedMessage),
+      }).unwrap();
+    } catch (error) {
+      setIsLoadingNewMessage(false);
+    }
+  };
+
+  useEffect(() => {
+    dispatch(setActiveChat(chat.buddyId));
+    setMessage('');
+  }, [chat.buddyId]);
+
+  useEffect(() => {
+    if (isLoadingNewMessage) {
+      setMessage('');
+      setIsLoadingNewMessage(false);
+    }
+  }, [chat.messages]);
```

Comment:
Instead of this effect, would it work if like this

```typescript
const handleMessageSend = async () => {
    if (!userId || isLoadingNewMessage) return;
    setIsLoadingNewMessage(true);
    const trimmedMessage = message.trim();
    try {
        await sendMessage({
         userId,
         message: toSendMessage(chat.buddyId, userId, trimmedMessage),
     }).unwrap();
       setMessage('');
     } catch (error) {
       console.error('Failed to send message:', error);
     } finally {
    setIsLoadingNewMessage(false);
  }
};
  ```

Reply:
```typescript
export const toSendMessage = (
  buddyId: string,
  userId: string,
  text: string,
) => ({
  recipient_id: buddyId,
  sender_id: userId,
  content: text.trim(),
  opened: false,
});
```

Reply:
Good points!

Reply:
I updated the `toSendMessage`, but your first suggestion won't work, because we only want to clear and enable the message field after the sent message is fetched from the api. Not yet when the post request returns ok.

Reply:
Ah okay,.. but wait what? 

I think my brain explode, or then i dont understand something:

If we have already two variables:

```typescript
const [sendMessage, { isLoading: isLoadingSendMessage }] =
    useSendMessageMutation(); 
    
    const isLoadingMessages = useAppSelector(
    selectIsLoadingBuddyMessages(chat?.buddyId),
  );
    ```
    
    Could we ditch the local-state and use those?
    
    ```typescript
   const ActiveWindow = () => {
  const { isTablet } = useGetLayoutMode();
  const dispatch = useAppDispatch();
  const [sendMessage, { isLoading: isLoadingSendMessage }] =
    useSendMessageMutation();
  const userId = useAppSelector(selectUserId);

  const activeChat = useAppSelector(selectActiveChat);
  const defaultChat = useAppSelector(selectDefaultChat);
  const chat = activeChat ?? defaultChat;

  const [message, setMessage] = useState('');

  const isLoadingMessages = useAppSelector(
    selectIsLoadingBuddyMessages(chat?.buddyId),
  );
  const isLoading = isLoadingMessages || isLoadingSendMessage;
  const isSendingDisabled = isLoadingNewMessage || message.trim().length < 1;

  const handleMessageSend = async () => {
    if (!userId || isLoading) return;

    try {
      await sendMessage({
        userId,
        message: toSendMessage(chat.buddyId, userId, message),
      }).unwrap();
      setMessage('')
    } catch (error) {
      console.error(error, 'error')
    }
  };

  useEffect(() => {
    dispatch(setActiveChat(chat.buddyId));
    setMessage('');
  }, [chat.buddyId]);

  return (
    chat && (
      <Container isTablet={isTablet}>
        <Header chat={chat} />
        <MessageList
          messageList={chat.messages}
          buddyId={chat.buddyId}
          status={chat.status}
          isLoading={isLoading}
        />
    ```

Reply:
```diff
diff --git a/src/features/Chat/components/ActiveWindow/index.tsx b/src/features/Chat/components/ActiveWindow/index.tsx
index 23a258f..c2318e5 100644
--- a/src/features/Chat/components/ActiveWindow/index.tsx
+++ b/src/features/Chat/components/ActiveWindow/index.tsx
@@ -7,12 +7,11 @@ import {
   selectDefaultChat,
 } from '@/features/Chat/selectors';
 import { selectUserId } from '@/features/Authentication/selectors';
-import { setActiveChat } from '@/features/Chat/chatSlice';
 import {
   toSendMessage,
   useSendMessageMutation,
 } from '@/features/Chat/chatPageApi';
-import { useAppDispatch, useAppSelector } from '@/store';
+import { useAppSelector } from '@/store';
 import { useGetLayoutMode } from '@/hooks/useGetLayoutMode';

 import {
@@ -30,7 +29,6 @@ import MessageList from './MessageList';

 const ActiveWindow = () => {
   const { isTablet } = useGetLayoutMode();
-  const dispatch = useAppDispatch();
   const [sendMessage, { isLoading: isLoadingSendMessage }] =
     useSendMessageMutation();
   const userId = useAppSelector(selectUserId);
:
-    pollingInterval: 2000,
+    pollingInterval: 5000,
   });

   return isAuthenticated ? (
diff --git a/src/features/Chat/components/ActiveWindow/index.tsx b/src/features/Chat/components/ActiveWindow/index.tsx
index 23a258f..c2318e5 100644
--- a/src/features/Chat/components/ActiveWindow/index.tsx
+++ b/src/features/Chat/components/ActiveWindow/index.tsx
@@ -7,12 +7,11 @@ import {

   return isAuthenticated ? (
diff --git a/src/features/Chat/components/ActiveWindow/index.tsx b/src/features/Chat/components/ActiveWindow/index.tsx
index 23a258f..c2318e5 100644
--- a/src/features/Chat/components/ActiveWindow/index.tsx
+++ b/src/features/Chat/components/ActiveWindow/index.tsx
@@ -7,12 +7,11 @@ import {
   selectDefaultChat,
 } from '@/features/Chat/selectors';
 import { selectUserId } from '@/features/Authentication/selectors';
-import { setActiveChat } from '@/features/Chat/chatSlice';
 import {
   toSendMessage,
   useSendMessageMutation,
 } from '@/features/Chat/chatPageApi';
-import { useAppDispatch, useAppSelector } from '@/store';
+import { useAppSelector } from '@/store';
 import { useGetLayoutMode } from '@/hooks/useGetLayoutMode';

 import {
@@ -30,7 +29,6 @@ import MessageList from './MessageList';

 const ActiveWindow = () => {
   const { isTablet } = useGetLayoutMode();
-  const dispatch = useAppDispatch();
   const [sendMessage, { isLoading: isLoadingSendMessage }] =
     useSendMessageMutation();
   const userId = useAppSelector(selectUserId);
@@ -45,36 +43,26 @@ const ActiveWindow = () => {
     selectIsLoadingBuddyMessages(chat?.buddyId),
   );
   const isLoading = isLoadingMessages || isLoadingSendMessage;
-
-  const [isLoadingNewMessage, setIsLoadingNewMessage] = useState(false);
-  const isSendingDisabled = isLoadingNewMessage || message.trim().length < 1;
+  const isSendingDisabled = isLoading || message.trim().length < 1;

   const handleMessageSend = async () => {
-    if (!userId || isLoadingNewMessage) return;
+    if (!userId || isLoading) return;

-    setIsLoadingNewMessage(true);
     try {
       await sendMessage({
         userId,
         message: toSendMessage(chat.buddyId, userId, message),
       }).unwrap();
+      setMessage('');
     } catch (error) {
-      setIsLoadingNewMessage(false);
+      console.error(error, 'error');
     }
   };

   useEffect(() => {
-    dispatch(setActiveChat(chat.buddyId));
     setMessage('');
   }, [chat.buddyId]);

-  useEffect(() => {
-    if (isLoadingNewMessage) {
-      setMessage('');
-      setIsLoadingNewMessage(false);
-    }
-  }, [chat.messages]);
-
   return (
     chat && (
       <Container isTablet={isTablet}>
@@ -88,7 +76,7 @@ const ActiveWindow = () => {
         {chat.status === 'ok' && (
           <MessageField
             handleSend={handleMessageSend}
-            isInputDisabled={isLoadingNewMessage}
+            isInputDisabled={isLoading}
             isSendDisabled={isSendingDisabled}
             message={message}
             onChange={setMessage}
```

This worked when I tested it. It would simplify the logic a bit, and less complexity is more robust

Reply:
Oh no yeah that is so confusing now... 😕 

isLoadingMessages = are the buddy's messages still loading
isLoadingSendMessage = we are waiting for an ok response after posting a new message... actually this one we probably don't need...
isLoadingNewMessage = we have sent a new message and are waiting for our polling system to update the message list

Reply:
Fixed! 💪